components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3587,4 +3585,33 @@ public void appendPolicy(String eligibilityQuery, String decisionQuery, String t
     public String getCondition(Policy policy) {
         return null;
     }
+
+    public String[] getPolicyNames(String username, String level) throws APIManagementException{
+        String[] policyNamesAPI = apiMgtDAO.getPolicyNames(level, username);
+        return policyNamesAPI;
+    }
+
+    public Policy[] getPolicies(String username, String level) throws APIManagementException{
+        Policy[] policies = apiMgtDAO.getPolicies(level, username);
+        return policies;
+    }
+
+    public void writeToFile(String content, String fileName) throws IOException {
+        File file =new File("repository/deployment/server/throttle-conf");
+        if(!file.exists()){
+            file.mkdir();
+        }
+        File writeFile = new File("repository/deployment/server/throttle-conf/"+fileName);
+        FileOutputStream fos = null;
+        fos = new FileOutputStream(file);
+        if (!writeFile.exists()) {
+            writeFile.createNewFile();
+        }
+
+        byte[] contentInBytes = content.getBytes();
+
+        fos.write(contentInBytes);
```

Comment:
Use finally method to close the streams if exception occurred the stream wont be  closed